src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LoweredDynamicOperationFactory.cs

```diff
@@ -549,6 +549,9 @@ internal static RefKind GetReceiverRefKind(BoundExpression loweredReceiver)
 
             switch (loweredReceiver.Kind)
             {
+                case BoundKind.Call:
```

Comment:
>case BoundKind.Call: [](http://example.com/codeflow?start=16&length=20)

Are we missing handling for other interesting scenarios? For example, I do not see explicit case for fields. We probably could run into a sequence (for example if we need to reorder arguments for a call), etc.
It looks like the purpose of this method is to determine whether we can take a reference to the receiver without storing the value in a temp. It feels fragile to have a helper that is used only for the purpose of dynamic rewrite, especially taking future language changes into account. Scenarios involving dynamic rewrite are easy to miss during testing. Therefore, I think we should either switch to a different existing helper, which is used more often (```Binder.HasHome```?), or we could try always pass value type receiver by reference and let emit layer to figure out if it needs a temp for that (it looks like that shouldn't change semantics of the operation).